### PR TITLE
Expose Modified Enchantment Level API

### PR DESCRIPTION
**Pull request by Swedz**

### CODE REVIEW

File: Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+diff --git a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+index ca097db904db96e81c0ea4b06b445a13e9333cfe..f0763d4ddc54531ca1387a8b23338bee7c6b4f66 100644
+--- a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
++++ b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+@@ -85,7 +85,19 @@ public class EnchantmentWrapper extends Enchantment {
+     public io.papermc.paper.enchantments.EnchantmentRarity getRarity() {
+         return getEnchantment().getRarity();
+     }
+-
++
```
Comment (Machine-Maker) [inline Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: Try and remove these diffs, all it is is changing the amount of whitespace on the line.

File: Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
++++ b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+@@ -85,7 +85,19 @@ public class EnchantmentWrapper extends Enchantment {
+     public io.papermc.paper.enchantments.EnchantmentRarity getRarity() {
+         return getEnchantment().getRarity();
+     }
+-
++    
++    @Override
++    public int getModifiedEnchantmentLevelMinimum(int level)
++    {
```
Comment (Machine-Maker) [inline Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: idk how style-nazi paper is, but like everywhere else the opening bracket is on the same line as the function definition

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
A few style changes throughout the diff

File: Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch (hunk 2 above)
Comment (Swedz) [reply Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: Yeah that's my bad. My IntelliJ is configured to work like that so it just automatically did it like that and I thought nothing of it.

Comment (Machine-Maker) [reply Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: Also, make sure to remove all whitespace on empty lines, if you don't, everyone elses IDEs will remove it automatically and itll just create diff anoyingness for others trying to modify the file without adding those extra whitespace removal changes.

Comment (Swedz) [reply Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: Sure thing. How can I change it? Do I need to make another commit on my fork? Sorry I'm new to PRs, trying to figure this all out.

Comment (Machine-Maker) [reply Spigot-API-Patches/0314-Expose-Modified-Enchantment-Level-API.patch]: So you make the changes in the API/Server folders, then just amend the previous commit to include those changes
* `git add .`
* `git commit --amend`
* rebuild patches
* commit changes to the main repo
* push to your remote branch

File: Spigot-Server-Patches/0752-Expose-Modified-Enchantment-Level-API.patch [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+diff --git a/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java b/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
+index c719c1c1f837753f61eb7be83b6ec2a214ab6819..93fc70e925860991723c6a98cde02763839f4257 100644
+--- a/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
++++ b/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
+@@ -58,11 +58,13 @@ public abstract class Enchantment {
+     public int getMaxLevel() {
+         return 1;
+     }
+-
++
```
Comment (Proximyst) [inline Spigot-Server-Patches/0752-Expose-Modified-Enchantment-Level-API.patch]: Remove such spaces

**Review by Proximyst — CHANGES_REQUESTED** [review]
There is no explanation as to what in the fuck a modified enchantment is. I have absolutely no idea, so Javadocs should be the only source of info I need.

File: Spigot-Server-Patches/0752-Expose-Modified-Enchantment-Level-API.patch (hunk 3 above)
Comment (Swedz) [reply Spigot-Server-Patches/0752-Expose-Modified-Enchantment-Level-API.patch]: Missed that one. My bad.

**Comment by Swedz** [comment]
> There is no explanation as to what in the fuck a modified enchantment is. I have absolutely no idea, so Javadocs should be the only source of info I need.

This information can quickly be found by googling it, I made sure before not including it. I asked about this in the paper-dev channel on discord and didn't get a response so I presumed it would just be brought up if needed.

**Comment by Machine-Maker** [comment]
I would link that document you linked in the discord in the javadoc
smth like
`@see <a href="that_link.com">Modified Enchant Values</a>`

**Comment by Swedz** [comment]
> I would link that document you linked in the discord in the javadoc
> smth like
> `@see <a href="that_link.com">Modified Enchant Values</a>`

Sure thing! Doing this right now.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
Needs to be rebased.  

I think the method names should be getMinimumModifiedEnchantmentLevel and getMaximumModifiedEnchantmentLevel

**Comment by Owen1212055** [comment]
Because it seems like this PR hasn't been updated in a while, this will be closed.
If you are still interested in this PR however, feel free to leave a comment and we can reopen it for you! 😄

**Pull request closed without merging**